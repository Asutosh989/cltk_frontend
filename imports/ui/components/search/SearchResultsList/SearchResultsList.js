import React from 'react';
import PropTypes from 'prop-types';
import Masonry from 'react-masonry-component/lib';

import WorkTeaser from '/imports/ui/components/works/WorkTeaser';

class SearchResultsList extends React.Component {

	isLoading = false
	worksCount = 0

	loadMore() {
		this.isLoading = true;
		this.props.loadMore();
	}

	renderWorks() {
		return this.props.works.map((work) => (
			<WorkTeaser
				key={work.id}
				work={work}
			/>
		));
	}

	render() {
		const masonryOptions = {
			// columnWidth : 400,
			// itemSelector: '.work-teaser',
			isFitWidth: true,
			transitionDuration: 300,
		};

		const works = this.props.works;

		if (this.worksLength !== works.length) {
			this.isLoading = false;
		}

		this.worksLength = works.length;

		return (
			<div className="works-list search-results-list">
				{works.length ?
					<Masonry
						options={masonryOptions}
						className="works-container works-container--grid row"
					>
						{this.renderWorks()}
					</Masonry>
				:
					<div className="works-container works-container--no-results">
						<p className="no-results">
							No results found for your query.
						</p>
					</div>
				}

				{this.props.hasMoreWorks ?
					<div>
					{this.isLoading ?
						<LoadingWell />
					:
						<a
							className="waves-effect waves-light btn-large"
							aria-label="View more"
							onClick={this.loadMore}
						>
							Load more
						</a>
					}
					</div>
				: '' }

			</div>
		);
	}
}

SearchResultsList.propTypes = {
	works: PropTypes.array.isRequired,
	loadMore: PropTypes.func.isRequired,
	hasMoreWorks: PropTypes.bool,
};

export default SearchResultsList;
