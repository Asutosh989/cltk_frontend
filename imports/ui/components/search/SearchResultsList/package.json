{
  "name": "SearchResultsList",
  "version": "0.0.1",
  "private": true,
  "main": "./SearchResultsList.js"
}
