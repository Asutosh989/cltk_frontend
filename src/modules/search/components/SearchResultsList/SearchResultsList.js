import React from 'react';
import PropTypes from 'prop-types';
import Masonry from 'react-masonry-component/lib';

import WorkTeaser from '/imports/ui/components/works/WorkTeaser';

class SearchResultsList extends React.Component {

	loadMore() {
		this.props.loadMore();
	}

	renderWorks() {
		return this.props.works.map((work) => (
			<WorkTeaser
				key={work.id}
				work={work}
			/>
		));
	}

	render() {
		const isLoading = false; // TODO: pass into props
		const masonryOptions = {
			// columnWidth : 400,
			// itemSelector: '.work-teaser',
			isFitWidth: true,
			transitionDuration: 300,
		};

		const works = this.props.works;

		return (
			<div className="works-list search-results-list">
				{works.length ?
					<Masonry
						options={masonryOptions}
						className="works-container works-container--grid row"
					>
						{this.renderWorks()}
					</Masonry>
				:
					<div className="works-container works-container--no-results">
						<p className="no-results">
							No results found for your query.
						</p>
					</div>
				}

				{this.props.hasMoreWorks ?
					<div>
						{isLoading ?
							<LoadingWell />
					:
							<a
								className="waves-effect waves-light btn-large"
								aria-label="View more"
								onClick={this.loadMore}
						>
							Load more
							</a>
					}
					</div>
				: '' }

			</div>
		);
	}
}

SearchResultsList.propTypes = {
	works: PropTypes.array.isRequired,
	loadMore: PropTypes.func.isRequired,
	hasMoreWorks: PropTypes.bool,
};

export default SearchResultsList;
