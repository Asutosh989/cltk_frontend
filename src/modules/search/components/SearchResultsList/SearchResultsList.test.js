
import React from 'react';
import { shallow } from 'enzyme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import SearchResultsList from './SearchResultsList';

describe('SearchResultsList', () => {
	it('renders correctly', () => {

		const wrapper = shallow(
			<MuiThemeProvider>
				<SearchResultsList
					works={[]}
					loadMore={() => {}}
					/>
			</MuiThemeProvider>
			);
		expect(wrapper).toBeDefined();
	});
});
