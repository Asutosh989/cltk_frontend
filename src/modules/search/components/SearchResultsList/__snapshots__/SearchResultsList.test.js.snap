// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsList renders correctly 1`] = `
<div
  className="works-list search-results-list"
>
  <div
    className="works-container works-container--no-results"
  >
    <p
      className="no-results"
    >
      No results found for your query.
    </p>
  </div>
  
</div>
`;
